### PR TITLE
Fix wakeup bug: Add wakeup token for ecc508 requirements

### PR DESCRIPTION
**Pull request by flyerink**
Fix wakeup bug: Add wakeup token for ecc508 requirements;
Remove I2C readout max byte limitation

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -48,6 +48,7 @@ an Erlang shell to communicate with the ecc:
     $ cd ecc508
     $ ./rebar3 shell
     $ {ok, Pid} = ecc508:start_link().
+    $ ecc508:wake(Pid).
```
Comment (madninja) [inline README.md:51]: having a wake here is useless since it is only valid for a few ms before the chip goes back to sleep

File: src/ecc508.erl [hunk 2]
```diff
@@ -83,7 +83,24 @@ stop(Pid) ->
 %% @doc Send a wake command to the ecc. This ensures that the ecc
 %% wakes up form its default sleep mode
 wake(Pid) ->
-    execute(Pid, command(wake)).
+    {ok, WakeHandle} = ecc508:start_link("i2c-1", 16#00),
```
Comment (madninja) [inline src/ecc508.erl:86]: This is incorrect.. you should not start the ec508 supervisor on every wake command

File: src/ecc508.erl [hunk 3]
```diff
@@ -83,7 +83,24 @@ stop(Pid) ->
...
+    BinWord = 00,
+    i2c:write(WakeHandle, <<BinWord:8/integer-unsigned>>),
```
Comment (madninja) [inline src/ecc508.erl:88]: Wy are you writing a wake, then an i2c stop and then issuing the same again in wait_for_response?

File: src/ecc508.erl [hunk 4]
```diff
@@ -895,7 +912,7 @@ read_response(_Pid, 0, Acc) ->
         false -> {error, ecc_checksum_failed}
     end;
 read_response(Pid, Length, Acc) ->
-    case i2c:read(Pid, min(Length, 32)) of
```
Comment (madninja) [inline src/ecc508.erl:898]: Why was this change made?

File: README.md (hunk 1 above)
Comment (flyerink) [reply README.md:51]: It is only for test the wake token

File: src/ecc508.erl (hunk 2 above)
Comment (flyerink) [reply src/ecc508.erl:86]: In ecc508 spec, it need wake up by write 0x00 to i2c at 100Khz before every command, or else chips may not wakeup to response the commands.

File: src/ecc508.erl (hunk 3 above)
Comment (flyerink) [reply src/ecc508.erl:88]: So wake up the ecc508 by write 0x00 to i2c. It is based on Microchip cryptoauthlib HAL source code.

File: src/ecc508.erl (hunk 4 above)
Comment (flyerink) [reply src/ecc508.erl:898]: No need to limit I2c read to 32 bytes. We could readout all the response directly.

File: README.md (hunk 1 above)
Comment (madninja) [reply README.md:51]: but it doesn't actually _do_ anything helpful in the README right? basically you want a wake followed by another command to do something

File: src/ecc508.erl (hunk 2 above)
Comment (madninja) [reply src/ecc508.erl:86]: yes but that does _not_ mean you `start_link` the ecc process on every wake.

File: src/ecc508.erl (hunk 3 above)
Comment (madninja) [reply src/ecc508.erl:88]: The `command(wake)` already does that write, doing it again is not adding anything useful here right?

File: src/ecc508.erl (hunk 4 above)
Comment (madninja) [reply src/ecc508.erl:898]: Per the ECC spec you can only do 4 or 32 byte reads from these zones. Have you been able to actually read more than 32 bytes?

Comment (flyerink) [reply src/ecc508.erl:898]: Yes, it working for more than 32bytes

File: src/ecc508.erl (hunk 3 above)
Comment (flyerink) [reply src/ecc508.erl:88]: The wake token you write is not correct according to the ecc608 spec, it need more than 60us low level from SDA pin, but not too enough for this. Some of ecc608 can't wake up to ACK the I2C request, especially in low temperature. So write 0x00 to I2C should better than older code.

File: src/ecc508.erl (hunk 4 above)
Comment (madninja) [reply src/ecc508.erl:898]: That seems to violate the spec of the ECC chip itself, no?

File: src/ecc508.erl (hunk 3 above)
Comment (madninja) [reply src/ecc508.erl:88]: 1. The problem is that you should not call start_link again. That fires up a new ecc erlang process for every packet which is not what you want here. 
2. command(wake) already implies sending a wakeup sequence. Perhaps we can fix it there? Your fix is for not sending enough 0x00 to the i2c to wake it up right? If so I'm agreeing, just not with the way you fixed it here

**Pull request closed without merging**